submissions/Nemooochka/memory-pair/scripts.js

```diff
@@ -0,0 +1,142 @@
+                </div>
+            </div>`;
+}
+
+function renderCards(arrayCards) {
+    const cards = document.createDocumentFragment();
+
+    arrayCards.map((elem) => {
+        let card = document.createElement('div');
+        card.classList.add('flip-container');
+        card.innerHTML = generateCard(elem);
+        cards.appendChild(card);
+    });
+
+    cardsWrapBlock.appendChild(cards);
+}
+
+function cardHover(event) {
+    const targetElem = event.target;
+
+    // setTimeout for Call Stack, execute after all previous code is done
+    setTimeout(function() {
+        if (targetElem.classList.contains("front")) {
+            // clear timer for resetting all hovered cards when user move to new card
+            clearTimeout(timerReset);
+
+            // show hovering of card smoothly
+            timerMouseOver = setTimeout(function () {
+                showedCard++;
+
+                // when 2 cards are shown and you move to the third, hide previous two
+                if (showedCard > 2) {
+                    showedCard = 1;
+                    prevCard = prevCardId = undefined;
+                    resetActiveCards();
+                }
+
+                currentCard = targetElem.closest('.flipper');
+                currentCard.classList.add('hover');
+                currentCardId = currentCard.getAttribute('data-pair');
+                currentCardIndex = currentCard.getAttribute('data-card');
+
+                if (currentCardId === prevCardId) {
+                    foundPairs++;
+                    // hide pairs smoothly
+                    setTimeout(function () {
+                        currentCard.closest('.flip-container').style.opacity = '0';
+                        prevCard.closest('.flip-container').style.opacity = '0';
+                    }, 500);
+
+                    // when all pair are founded
+                    if (foundPairs === cards.length) {
+                        foundPairs = 0;
+                        setTimeout(function () {
+                            cardsWrapBlock.innerHTML = '' +
+                                '<div class="congratulations">' +
+                                '<p>Wow! You are really good at it. Congratulations! :)</p>' +
+                                '<button class="btn btn-new_game">One more time?</button>' +
+                                '</div>';
+                            newGameBtn = cardsWrapBlock.querySelector('.btn-new_game');
+                            newGameBtn.addEventListener("click", () => {
+                                cardsWrapBlock.innerHTML = '';
+                                renderCards(Shuffle(pairArray));
+                            });
+                        }, 500);
+                    }
+                } else {
+                    prevCard = currentCard;
+                    prevCardId = currentCardId;
+                    prevCardIndex = currentCardIndex;
+                }
+            }, 500);
```

Comment:
@MaxKalashnyk, because I use this setTimeout to reset on mouseout at the end of script file cardsWrapBlock.addEventListener("mouseout", () => clearTimeout(timerMouseOver));

This setTimeout needs to be available for the whole project. I tried to find a way to more setTimout as const to the beginning of the file, where other variables are but didn't get an idea. Could you help how to do it right?